### PR TITLE
Feature/csi 3357 encode vdisk volumehandle

### PR DESCRIPTION
**Pull request by ArbelNathan**

### CODE REVIEW

File: controller/controller_server/config.py [hunk 1]
```diff
@@ -37,6 +37,7 @@
 PARAMETERS_CAPACITY_DELIMITER = "="
 PARAMETERS_CAPABILITIES_DELIMITER = "="
 PARAMETERS_OBJECT_ID_DELIMITER = ":"
+PARAMETERS_INTERNAL_ID_DELIMITER = ";"
```
Comment (oriyarde) [inline controller/controller_server/config.py:40]: I think we now need better names to distinguish between the delimiters.
how about:
```suggestion
PARAMETERS_OBJECT_ID_INFO_DELIMITER = ":"
PARAMETERS_OBJECT_IDS_DELIMITER = ";"
```

File: controller/tests/controller_server/csi_controller_server_test.py [hunk 2]
```diff
@@ -399,10 +420,12 @@ def _test_create_volume_succeeds(self, storage_agent, expected_pool=pool):
         self.mediator.create_volume.assert_called_once_with(volume_name, 10, None, expected_pool)
         self.assertEqual(response_volume.volume.content_source.volume.volume_id, '')
         self.assertEqual(response_volume.volume.content_source.snapshot.snapshot_id, '')
+        return response_volume
```
Comment (oriyarde) [inline controller/tests/controller_server/csi_controller_server_test.py:423]: ?

Comment (oriyarde) [inline controller/tests/controller_server/csi_controller_server_test.py:423]: consider passing `expected_volume_id` instead

File: controller/controller_server/test_settings.py [hunk 3]
```diff
@@ -4,6 +4,7 @@
 pool = "pool1"
 space_efficiency = "thin"
 volume_name = "volume"
+object_id = "0"
```
Comment (oriyarde) [inline controller/controller_server/test_settings.py:7]: consider using more indicative name and value, e.g.
```suggestion
internal_id = "internal-id"
```

File: controller/tests/controller_server/csi_controller_server_test.py [hunk 4]
```diff
@@ -923,13 +947,26 @@ def test_delete_volume_has_snapshots(self):
 
     @patch("controller.array_action.array_mediator_xiv.XIVArrayMediator.delete_volume")
     @patch("controller.controller_server.csi_controller_server.get_agent")
-    def test_delete_volume_succeeds(self, storage_agent, delete_volume):
+    def _test_delete_volume_succeeds(self, volume_id, storage_agent, delete_volume):
         storage_agent.return_value = self.storage_agent
         delete_volume.return_value = Mock()
+        self.request.snapshot_id = volume_id
```
Comment (oriyarde) [inline controller/tests/controller_server/csi_controller_server_test.py:953]: snapshot?

File: controller/tests/controller_server/utils_test.py [hunk 5]
```diff
@@ -351,14 +351,37 @@ def test_validate_unpublish_volume_request(self, validate_secrets):
 
         utils.validate_unpublish_volume_request(request)
 
+    def _test_get_volume_id_info(self, object_id, system_id=None, internal_id=None):
+        system_id_field = ':' + system_id if system_id else ''
+        id_field = internal_id + ';' + object_id if internal_id else object_id
```
Comment (oriyarde) [inline controller/tests/controller_server/utils_test.py:356]: IMO format is more readable, but if you prefer it like this, I will resolve

Comment (oriyarde) [inline controller/tests/controller_server/utils_test.py:356]: I think `ids_field` would be better here

File: node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go [hunk 6]
```diff
@@ -73,6 +73,7 @@ const (
 	multipathdCmd               = "multipathd"
 	multipathCmd                = "multipath"
 	VolumeIdDelimiter           = ":"
+	VolumeStorageIdDelimiter    = ";"
```
Comment (oriyarde) [inline node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go:76]: IMO `Ids` would be better here

File: controller/controller_server/config.py (hunk 1 above)
Comment (ArbelNathan) [reply controller/controller_server/config.py]: done

File: controller/controller_server/test_settings.py (hunk 3 above)
Comment (ArbelNathan) [reply controller/controller_server/test_settings.py]: done

File: controller/tests/controller_server/csi_controller_server_test.py (hunk 4 above)
Comment (ArbelNathan) [reply controller/tests/controller_server/csi_controller_server_test.py]: oops
done

File: controller/tests/controller_server/utils_test.py (hunk 5 above)
Comment (ArbelNathan) [reply controller/tests/controller_server/utils_test.py]: done

Comment (ArbelNathan) [reply controller/tests/controller_server/utils_test.py]: done

File: node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go (hunk 6 above)
Comment (ArbelNathan) [reply node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go]: done

File: controller/controller_server/test_settings.py (hunk 3 above)
Comment (oriyarde) [reply controller/controller_server/test_settings.py]: same for [here](https://github.com/IBM/ibm-block-csi-driver/blob/aeb2553d0681192f370ea1520186d38572486976/controller/tests/array_action/xiv/array_mediator_xiv_tests.py#L75), [here](https://github.com/IBM/ibm-block-csi-driver/blob/aeb2553d0681192f370ea1520186d38572486976/controller/tests/array_action/svc/array_mediator_svc_test.py#L273), etc...

Comment (ArbelNathan) [reply controller/controller_server/test_settings.py]: those are storage objects id parameter, e.g. `cli_volume.id`

Comment (oriyarde) [reply controller/controller_server/test_settings.py]: oh, right

Comment (oriyarde) [reply controller/controller_server/test_settings.py]: then how about:
```suggestion
object_id = "object-id"
```

Comment (ArbelNathan) [reply controller/controller_server/test_settings.py]: `Shadows built-in name 'id' `
done `object_id`..

**Review by zingero — APPROVED** [review]

File: controller/controller_server/test_settings.py (hunk 3 above)
Comment (oriyarde) [reply controller/controller_server/test_settings.py]: sorry, I meant `'id': 'test-id'`

Comment (ArbelNathan) [reply controller/controller_server/test_settings.py]: ok, done..

**Review by oriyarde — APPROVED** [review]
LGTM

**Pull request merged**